sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -401,9 +411,98 @@ def _get_environment(
         env = self._environment_from_row(row)
         return env
 
+    def _backup_table(self, schema_version: int, table_name: str) -> None:
+        data = self.engine_adapter.fetchdf(exp.select("*").from_(table_name))
+        self.engine_adapter.insert_append(
+            self.backups_table,
+            next(
```

Comment:
Wouldn't it be a little too much storing entire tables as a single row? Should we consider just copying tables instead?

Reply:
i was suggesting storing one column per row, but we could copy the tables

Reply:
I gave it some thought and I think copying the tables and replacing them on rollback has the advantage of us not having to worry about a migration that alters one or more of the state tables.